src/object/cloud_particle_system.cpp

```diff
@@ -138,15 +193,34 @@ void CloudParticleSystem::update(float dt_sec)
   // Clear dead clouds.
   // Iterate through the vector backwards to avoid affecting the index of elements
   // after removal, preventing buggy behavior.
-  for (int i = static_cast<int>(particles.size()) - 1; i >= 0; --i) {
+  for (int i = static_cast<int>(particles.size()) - 1; i >= 0; --i)
+  {
     auto particle = dynamic_cast<CloudParticle*>(particles.at(i).get());
     
     if (particle->target_alpha == 0.f && particle->target_time_remaining == 0.f)
       particles.erase(particles.begin()+i);
   }
 }
 
-int CloudParticleSystem::add_clouds(int amount, float fade_time)
+void
+CloudParticleSystem::apply_fog_effect(DrawingContext& context)
+{
+  float opacity = fog_max_value * (m_current_amount - fog_start_amount) / (max_amount - fog_start_amount);
+  if (opacity < 0.f)
```

Comment:
Maybe use `math::clamp`? @MatusGuy Isn't that right?

Reply:
Yeah